tools/ion-bench/src/main.cpp

```diff
@@ -0,0 +1,260 @@
+#include "benchmark/benchmark.h"
+#include <string.h>
+#include <iostream>
+#include <map>
+#include <functional>
+
+#include <argtable3.h>
+
+#include "ion/ionc.h"
+#include "json/yy.h"
+#include "json/json-c.h"
+#include "msgpack/msgpack.h"
+#include "cbor/libcbor.h"
+#include "benchmarks.h"
+
+#include "common.h"
+#ifdef TRACK_MEMORY
+#  include "memory.h"
+#endif
+
+Library *libraries[] = {
+   new ion::IonC<ion::Format::Binary>(),
+   new ion::IonC<ion::Format::Text>(),
+   new ion::IonC<ion::Format::JSON>(),
+   new yyjson::YYJson(),
+   new jsonc::JsonC(),
+   new msgpack::MsgPackC(),
+   new libcbor::LibCBOR()
+};
+
+Library *find_library(char const *l) {
+   for (int i=0; i < sizeof(libraries)/sizeof(libraries[0]); i++) {
+      if (!strcmp(libraries[i]->name(), l))
+         return libraries[i];
+   }
+   return nullptr;
+}
+
+typedef std::function<int(benchmark::State &st, Library*, const char *, bool)> BenchmarkFunc;
+std::map<std::string, BenchmarkFunc> benchmarks = {
+   { "deserialize_all", deserialize_all() },
+   { "serialize_all",   serialize_all() },
+};
+
+template <size_t N>
+struct data_name {
+   char _chars[N+1] = {};
+
+   constexpr data_name(char const *s) {
+      for (unsigned i = 0; i < N; ++i)
+         _chars[i] = s[i];
+   }
+
+   constexpr operator char const*() const { return _chars; }
+   constexpr size_t length() const { return N; }
+};
+
+template<size_t N>
+data_name(char const (&)[N]) -> data_name<N - 1>;
+
+/*
+ * Deserialization benchmark.
+ */
+// template <typename T, data_name dataset, Format format>
+// void deser(benchmark::State &state) {
```

Comment:
Looks like this stuff moved to benchmarks.h. Let's delete the commented code if it's not needed.